core/include/gnuradio-4.0/WaitStrategy.hpp

```diff
@@ -245,10 +245,12 @@ class SpinWait {
             yieldProcessor();
         }
     }
-#if not defined(__EMSCRIPTEN__) && not defined(__APPLE__)
-    static void yieldProcessor() noexcept { asm volatile("rep\nnop"); }
-#else
+#if defined( __EMSCRIPTEN__)
     static void yieldProcessor() noexcept { std::this_thread::sleep_for(std::chrono::milliseconds(1)); }
+#elif defined(DISRUPTOR_CPU_ARM)
+    static void yieldProcessor() noexcept {  asm volatile("yield"); }
+#else
+    static void yieldProcessor() noexcept { asm volatile("rep\nnop"); }
```

Comment:
Out of curiosity, are the tests working for you? I've tried this a half a year or soago on my M3 mac (Linux VM), but the tests took forever (like 30 minutes) to run, with many failing, so I didn't pursue this further, at the time.

Reply:
I am cross-compiling using Yocto.  There were a few other issues that I worked around, but the fixes for them are too bad to use.  I will probably create some issues for them soon.  There are things like mtune native in some Cmakefiles for example.  I also have a powerful laptop for building,  I had to add a 64GB swap drive, in addition to 64GB of RAM to complete the build.

Reply:
Actually I ran the tests again, and it looks much better now, only one failing test (qa_buffer), and all of them completing in a somewhat reasonable time (120s).